orderBook.js

```diff
@@ -1,5 +1,33 @@
 function reconcileOrder(existingBook, incomingOrder) {
-  throw new Error('Not Implemented Yet')
+  if (existingBook.length === 0) {
+    existingBook.push(incomingOrder)
+  } else {
+    existingBook.forEach(existingOrders => {
+      if (existingOrders.type !== incomingOrder.type) {
+        if (existingOrders.quantity === incomingOrder.quantity) {
+          if (existingOrders.price < incomingOrder.price) {
+            existingBook.push(incomingOrder)
+          } else {
+            existingBook.pop(existingOrders)
+          }
+        } else if (existingOrders.price === incomingOrder.price) {
+          if (incomingOrder.quantity < existingOrders.quantity) {
+            existingOrders.quantity =
+              existingOrders.quantity - incomingOrder.quantity
+          } else {
+            incomingOrder.quantity =
+              incomingOrder.quantity - existingOrders.quantity
+            existingBook.pop(existingOrders) && existingBook.push(incomingOrder)
+          }
+        } else {
+          existingBook.push(incomingOrder)
+        }
+      } else {
+        existingBook.push(incomingOrder)
+      }
+    })
+  }
+  return existingBook
```

Comment:
This is a bit "pyramid of doom" which is what we call this many nested if/else statements. We would definitely want subfunctions for this. This structure would be very tough to properly test and maintain.